### PR TITLE
fix(clickhouse handler): try catch error before response.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. return runtime error message, instead of disconnect.
2. returning data in 2 steps, 
   1. try to collect 2 blocks from the result data_stream,  if there is an error,  return 500 directly. most runtime errors can be cached here, at least those cases in the current logic test
   2. streaming the others

see the issues below：

Fixes https://github.com/datafuselabs/databend/issues/6262
https://github.com/datafuselabs/databend/issues/6993

### CODE REVIEW

**Comment by youngsofun** [comment]
test_sqllogic_ydb_standalone_linux (pull_request) 

```
INFO:log:Suite file:suites/ydb/select1-4.test pass!
INFO:log:Suite file:suites/ydb/select1-5.test pass!
INFO:log:Suite file:suites/ydb/select2-1.test pass!
INFO:log:Suite file:suites/ydb/select2-2.test pass!
INFO:log:Suite file:suites/ydb/select2-3.test pass!
Error: The operation was canceled.

```

@ZeaLoVe  timeout?

**Comment by youngsofun** [comment]
Sun, 07 Aug 2022 15:29:49 GMT
INFO:log:Suite file:suites/ydb/select3-4.test pass!
1072
Sun, 07 Aug 2022 15:34:36 GMT
Error: The operation was canceled


maybe timeout because some cases are not skiped anymore？

**Review by sundy-li — APPROVED** [review]

File: query/src/servers/http/clickhouse_handler.rs [hunk 1]
```diff
@@ -139,18 +139,45 @@ async fn execute(
             rb
         }
     };
+
+    let mut blocks: Vec<Result<Vec<u8>>> = vec![];
+    for _ in 0..2 {
```
Comment (Xuanwo) [inline query/src/servers/http/clickhouse_handler.rs:144]: How about leaving some comments to explain why `0..2` here?

Comment (youngsofun) [reply query/src/servers/http/clickhouse_handler.rs:144]: OK

**Comment by ZeaLoVe** [comment]
I think we need timeout for execute single SQL.

**Review by zhang2014 — APPROVED** [review]

**Pull request merged**